src/pyOpenMS/pyopenms/addons/ximparquetfile.py

```
@@ -0,0 +1,337 @@
+    Returns
+    -------
+    list
+        List of column names.
+    """
+    return self.getColumns()
+
+
+@addon("XIMParquetFile")
+def get_data_dict(self, explode=False, precursor_id=-1, transition_id=-1,
+                  modified_sequence="", precursor_charge=-1, product_charge=-1,
+                  ms_level=-1, run_id=-1, mobilogram_type="", feature_id=-1, feature_rt=-1.0, filter=""):
```

Ruff (0.15.5)

[error] Function argument `filter` is shadowing a Python builtin

(A002)